server.mjs

```diff
@@ -37,53 +39,60 @@ app.get("/forecast", async (_, res) => {
     const data = await getDataFromLogs();
     res.writeHead(200, { "Content-Type": "application/json" });
     res.end(JSON.stringify(data), null, 2);
+    return;
   } catch {
     res.status(500).send("Internal Server Error");
+    return;
   }
 });
 
-// TODO: .get('/cityForecast/:cityName', (...) => {...})     (query parameter > route parameter)
 app.get("/cityForecast/:cityName", async (req, res) => {
+  const cityName = req.params.cityName.toLowerCase();
+  if (!cityName) {
+    res.status(400).json({ error: "No city query parameter provided" });
+    return;
+  }
   try {
-    const cityName = req.params.cityName.toLowerCase();
-    if (cityName) {
-      const cityForecast = await getForecastForCityIes(cityName);
-      res.status(200).json(cityForecast);
-    } else {
-      res.status(400).json({ error: "No city query parameter provided" });
-    }
+    const cityForecast = await getForecastForCityIes(cityName);
+    res.status(200).json(cityForecast);
+    return;
   } catch (error) {
     console.log(error);
     res.status(500).send("Internal Server Error");
+    return;
   }
 });
 
-app.get("/addCity", (req, res) => {
+app.get("/addCity", (_, res) => {
   res.sendFile(path.join(__dirname, "templates/add_city.html"));
+  return;
 });
```

Comment:
We don't need this EP on a back-end side
html - it's a static files, that might be reached directly (just drag and drop the file into browser tab)

If you wanna create a merged front-end back-end platform, you would need to respond with html data instead of json and create an index.html file that would contain whatever you need and dynamically fulfill with back-end data during requests
I wouldn't recommend to do so, at least for now, but it's up to you